Importación librerias necesarías para realizar el EDA.

In [2]:
# Tratamiento de datos
import pandas as pd
import numpy as np

1. EDA del primer dataset ('bank-additional.csv'):

In [4]:
# Cargar el archivo csv 'bank-additional.csv':
bank = pd.read_csv('Data/bank-additional.csv',index_col = 0) # index_col = 0 para omitir el indice que pone pandas al cargar el datraframe

# Visualizar nombre de las columnas y las primeras 5 filas para comprobar que se ha cargado correctamente:
bank.head() # 5 primeras filas

age        job  marital    education  default  housing  loan    contact  \
0   NaN  housemaid  MARRIED     basic.4y      0.0      0.0   0.0  telephone   
1  57.0   services  MARRIED  high.school      NaN      0.0   0.0  telephone   
2  37.0   services  MARRIED  high.school      0.0      1.0   0.0  telephone   
3  40.0     admin.  MARRIED     basic.6y      0.0      0.0   0.0  telephone   
4  56.0   services  MARRIED  high.school      0.0      0.0   1.0  telephone   

   duration  campaign  ...  emp.var.rate  cons.price.idx cons.conf.idx  \
0       261         1  ...           1.1          93,994         -36,4   
1       149         1  ...           1.1          93,994         -36,4   
2       226         1  ...           1.1          93,994         -36,4   
3       151         1  ...           1.1          93,994         -36,4   
4       307         1  ...           1.1          93,994         -36,4   

   euribor3m nr.employed   y                date latitude longitude  \
0      4,857        5191  no       2-agosto-2019   41.495   -71.233   
1        NaN        5191  no  14-septiembre-2016   34.601   -83.923   
2      4,857        5191  no     15-febrero-2019   34.939   -94.847   
3        NaN        5191  no   29-noviembre-2015   49.041   -70.308   
4        NaN        5191  no       29-enero-2017   38.033  -104.463   

                                    id_  
0  089b39d8-e4d0-461b-87d4-814d71e0e079  
1  e9d37224-cb6f-4942-98d7-46672963d097  
2  3f9f49b5-e410-4948-bf6e-f9244f04918b  
3  9991fafb-4447-451a-8be2-b0df6098d13e  
4  eca60b76-70b6-4077-80ba-bc52e8ebb0eb  

[5 rows x 23 columns]

In [5]:
# Dimensiones del set de datos:
print(f'Número de filas: {bank.shape[0]}\nNúmero de columnas: {bank.shape[1]}') 

Número de filas: 43000
Número de columnas: 23


In [7]:
# Visualizar el nombre de las columnas con el tipo de datos y si tienen nulos.
bank.info()

<class 'pandas.core.frame.DataFrame'>
Index: 43000 entries, 0 to 27570
Data columns (total 23 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             37880 non-null  float64
 1   job             42655 non-null  object 
 2   marital         42915 non-null  object 
 3   education       41193 non-null  object 
 4   default         34019 non-null  float64
 5   housing         41974 non-null  float64
 6   loan            41974 non-null  float64
 7   contact         43000 non-null  object 
 8   duration        43000 non-null  int64  
 9   campaign        43000 non-null  int64  
 10  pdays           43000 non-null  int64  
 11  previous        43000 non-null  int64  
 12  poutcome        43000 non-null  object 
 13  emp.var.rate    43000 non-null  float64
 14  cons.price.idx  42529 non-null  object 
 15  cons.conf.idx   43000 non-null  object 
 16  euribor3m       33744 non-null  object 
 17  nr.employed     43000 non-null  obje

**Columnas con nulos**: age, job, marital, education, default, housing, loan, cons.price.idx, euribor3m, date


**Columnas con tipo de datos erroneo**:

cons.price.idx (texto) convertir a decimal
cons.conf.idx (texto) conveertir a decimal
euribor3m (texto) convertir a decimal
nr.employed (texto) convertira entero
date (texto) convertir a fecha
default (decimal) convertir a entero
housing (decimal) convertir a entero
loan (decimal) convertir a entero

Descripción estadistica de las columnas numéricas:

In [8]:
bank.describe().T # descripción estadistica de columnas numéricas por defecto, con .T muestra las transposición de las mismas

count        mean         std      min        25%       50%  \
age           37880.0   39.977112   10.437957   17.000   32.00000   38.0000   
default       34019.0    0.000088    0.009390    0.000    0.00000    0.0000   
housing       41974.0    0.535998    0.498708    0.000    0.00000    1.0000   
loan          41974.0    0.155620    0.362499    0.000    0.00000    0.0000   
duration      43000.0  257.739279  258.666033    0.000  102.00000  179.0000   
campaign      43000.0    2.567233    2.772294    1.000    1.00000    2.0000   
pdays         43000.0  962.330953  187.260394    0.000  999.00000  999.0000   
previous      43000.0    0.174023    0.497366    0.000    0.00000    0.0000   
emp.var.rate  43000.0    0.077128    1.573898   -3.400   -1.80000    1.1000   
latitude      43000.0   36.856697    7.225948   24.396   30.61475   36.7610   
longitude     43000.0  -95.939067   16.752282 -124.997 -110.49425  -95.8995   

                    75%       max  
age            47.00000    98.000  
default         0.00000     1.000  
housing         1.00000     1.000  
loan            0.00000     1.000  
duration      319.00000  4918.000  
campaign        3.00000    56.000  
pdays         999.00000   999.000  
previous        0.00000     7.000  
emp.var.rate    1.40000     1.400  
latitude       43.11325    49.384  
longitude     -81.42775   -66.937

La información de este dataframe muestra:

- default, housing y loan tienen unos y ceros
- pdays tiene una anomalia en los valores teniendo 999 en los días que han pasado desde el último contacto, posiblemente sea el valor arbitrario que le daban en el banco a los clientes con los que no se había contactado
- duration puede que tenga outliers
- campaign puede que tenga outliers
- latitude y longitude son coordenadas y no es necesario sacar información estadistica de ellas

Descripción estadistica de las columnas categóricas:

In [9]:
bank.describe(include ='O').T # para las columnas categóricas, la "O" equivale a object

count unique                                   top   freq
job             42655     11                                admin.  10873
marital         42915      3                               MARRIED  25999
education       41193      7                     university.degree  12722
contact         43000      2                              cellular  27396
poutcome        43000      3                           NONEXISTENT  37103
cons.price.idx  42529     26                                93,994   7938
cons.conf.idx   43000     26                                 -36,4   8020
euribor3m       33744    309                                 4,857   2287
nr.employed     43000     11                                5228,1  16980
y               43000      2                                    no  38156
date            42752   1825                       28-febrero-2018     95
id_             43000  43000  993bbbd6-4dbc-4a40-a408-f91f8462bee6      1

id_ no se repite, por lo que es el identificador único de cada cliente.

Elementos únicos de las columnas categóricas y la cantidad de veces que se repiten: